rpmio/rpmio.c

```diff
@@ -1116,6 +1116,28 @@ int Fseek(FD_t fd, off_t offset, int whence)
     return rc;
 }
 
+int Fsync(FD_t fd)
+{
+    int rc = 0, ec = 0;
+
+    if (fd == NULL)
+	return -1;
+
+    fd = fdLink(fd);
+    for (FDSTACK_t fps = fd->fps; fps != NULL; fps = fps->prev) {
+	if (fps->fdno >= 0) {
+            rc = fsync(fps->fdno);
+	    if (ec == 0 && rc)
+		ec = rc;
+	}
+
+	/* Leave freeing the last one after stats */
+	if (fps->prev == NULL)
+	    break;
```

Comment:
This condition is quite clearly a leftover from copy-pasting Fclose(). It's probably not harmful to the cause here but it's also unnecessary and the comment is just plain wrong in this context.
The also copy-pasted fdLink() earlier is more harmful, it'll cause a file descriptor (and memory) leak since it's incrementing the reference count without ever decrementing it. 

These were kinda implied between the lines in the "just walk fps->prev" (instead of using copy-pasted Fclose code), sorry for not being more clear about it.